src/components/ui/addressform/addressform.main.jsx

```diff
@@ -0,0 +1,328 @@
+    }
+    fetchAddressData(addressLink) {
+        login().then(() => {
+            fetch(addressLink, {
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Authorization': localStorage.getItem(Config.cortexApi.scope + '_oAuthToken')
+                }
+            })
+                .then(res => res.json())
+                .then(res => {
+                    this.setState({
+                        firstName: res['name']['given-name'],
+                        lastName: res['name']['family-name'],
+                        address: res['address']['street-address'],
+                        extendedAddress: res['address']['extended-address'] ? res['address']['extended-address'] : "",
+                        city: res['address']['locality'],
+                        country: res['address']['country-name'],
+                        subCountry: res['address']['region'],
+                        postalCode: res['address']['postal-code']
+                    })
+                })
+        });
+    }
+    componentDidMount() {
+        this.fetchGeoData();
+        if (this.props.location.state && this.props.location.state.address) {
+            this.fetchAddressData(this.props.location.state.address);
+        }
+    }
+    setFirstName(event) {
+        this.setState({ firstName: event.target.value });
+    }
+    setLastName(event) {
+        this.setState({ lastName: event.target.value });
+    }
+    setAddress(event) {
+        this.setState({ address: event.target.value });
+    }
+    setExtendedAddress(event) {
+        this.setState({ extendedAddress: event.target.value });
+    }
+    setCity(event) {
+        this.setState({ city: event.target.value });
+    }
+    setCountry(event) {
+        this.setState({ country: event.target.value });
+    }
+    setSubCountry(event) {
+        this.setState({ subCountry: event.target.value });
+    }
+    setPostalCode(event) {
+        this.setState({ postalCode: event.target.value });
+    }
+    renderCountries() {
+        if (this.state.geoData) {
+            const sortedCountries = [].concat(this.state.geoData['_element'])
+                .sort((a, b) => {
+                    if (a['display-name'] > b['display-name']) {
+                        return 1;
+                    } else {
+                        return -1;
+                    }
+                });
+            return (
+                sortedCountries.map((country) => {
+                    return (
+                        <option key={country['name']} value={country['name']}>{country['display-name']}</option>
+                    );
+                })
+            );
+        }
+    }
+    renderSubCountries() {
+        if (this.state.country && this.state.geoData) {
+            const countryData = this.state.geoData['_element'].find((element) => { return element['name'] === this.state.country });
+            if (countryData['_regions'][0]['_element']) {
```

Comment:
Do countries really get returned unsorted? You have to sort them in alphabetical order yourself?

Reply:
yes, cortex returns unsorted, but I wanted them to be displayed in alphabetical order in the dropdowns so they are easier to select.